src/SDKs/Azure.Core.Identity/data-plane/Azure.Core.Identity/IdentityClient.cs

```diff
@@ -0,0 +1,155 @@
+        public IdentityClientOptions()
+        {
+            AuthorityHost = DefaultAuthorityHost;
+            LoggingPolicy = Core.Pipeline.Policies.LoggingPolicy.Shared;
+            RetryPolicy = new ExponentialRetryPolicy()
+            {
+                Delay = TimeSpan.FromMilliseconds(800),
+                MaxRetries = 3
+            };
+        }
+    }
+
+    internal class IdentityClient
+    {
+        private readonly IdentityClientOptions _options;
+        private readonly HttpPipeline _pipeline;
+        
+
+        public IdentityClient(IdentityClientOptions options = null)
+        {
+            _options = options ?? new IdentityClientOptions();
+
+            _pipeline = HttpPipeline.Build(options,
+                    options.ResponseClassifier,
+                    options.RetryPolicy,
+                    ClientRequestIdPolicy.Singleton,
+                    options.LoggingPolicy,
+                    BufferResponsePolicy.Singleton);
+        }
+
+        public async Task<AuthenticationResponse> AuthenticateAsync(string tenantId, string clientId, string clientSecret, IEnumerable<string> scopes, CancellationToken cancellationToken = default)
+        {
+            Uri authUri = new Uri(_options.AuthorityHost + tenantId + "/oauth2/token");
+
+            using (var request = _pipeline.CreateRequest())
+            {
+                var body = new Dictionary<string, string>() {
+                    { "response_type", "token" },
+                    { "grant_type", "client_credentials" },
+                    { "client_id", clientId },
+                    { "client_secret", clientSecret },
+                    { "scopes", string.Join(" ", scopes) }
+                };
+
+                request.SetRequestLine(HttpPipelineMethod.Put, authUri);
+
+                ReadOnlyMemory<byte> content = Serialize(body);
+
+                request.Content = HttpPipelineRequestContent.Create(content);
+
+                var response = await _pipeline.SendRequestAsync(request, cancellationToken).ConfigureAwait(false);
+
+                if (response.Status == 200 || response.Status == 201)
+                {
+                    var result = await DeserializeAsync(response.ContentStream, cancellationToken).ConfigureAwait(false);
+
+                    return new Response<AuthenticationResponse>(response, result);
+                }
+
+                else throw new RequestFailedException(response);
+            }
+        }
+
+        private ReadOnlyMemory<byte> Serialize(Dictionary<string,string> dict)
+        {
+            var buff = new byte[1024];
+
+            var writer = new FixedSizedBufferWriter(buff);
+
+            var json = new Utf8JsonWriter(writer);
+
+            json.WriteStartObject();
+
+            foreach(var prop in dict)
+            {
+                json.WriteString(prop.Key, prop.Value);
+            }
+
+            json.WriteEndObject();
+
+            return buff.AsMemory(0, (int)json.BytesWritten);
```

Comment:
The latest `Utf8JsonWriter` is disposable and needs to be disposed/surrounded with using. But it looks like you are referencing an older version.